src/riscv/lib/src/pvm/linux/parameters.rs

```diff
@@ -77,21 +79,155 @@ impl TryFrom<u64> for MainThreadId {
-#[derive(Debug, Clone, Copy)]
-pub struct ProcessId;
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct ProcessId(pub u64);
 
 impl TryFrom<u64> for ProcessId {
     type Error = Error;
 
     fn try_from(value: u64) -> Result<Self, Self::Error> {
         // We only support the single hart or a null value
         match value {
-            0 | 1 => Ok(ProcessId),
+            id @ 0 | id @ 1 => Ok(ProcessId(id)),
             _ => Err(Error::Search),
         }
     }
 }
 
+/// Hard limit on CPU time (s)
+const RLIMIT_CPU: u64 = u64::MAX;
+
+/// Hard limit for maximum file size (we don't support writing to a filesystem)
+const RLIMIT_FSIZE: u64 = 0;
+
+/// Hard limit for core dumps (we don't support core dumps)
+const RLIMIT_CORE: u64 = 0;
+
+/// Hard limit on processes (we only support one)
+const RLIMIT_NPROC: u64 = 1;
+
+/// Hard limit on the number of file descriptors that a system call can work with
+///
+/// We also use this constant to implictly limit how much memory can be associated with a system
+/// call. For example, `ppoll` takes a pointer to an array of `struct pollfd`. If we don't limit
+/// the length of that array, then we might read an arbitrary amount of memory. This impacts the
+/// proof size dramatically as everything read would also be in the proof.
+pub const RLIMIT_NOFILE: u64 = 512;
+
+/// Hard limit on the number of flock locks plus fcntl locks. Unused by us (and we only support one hart and no filesystem)
+const RLIMIT_LOCKS: u64 = 0;
+
+/// Hard limit on the number of signals for a user ID. Signals are not currently implemented.
+const RLIMIT_SIGPENDING: u64 = 0;
+
+/// Hard limit on the number of bytes that may be allocated for message queues. Message queues are not currently implemented.
+const RLIMIT_MSGQUEUE: u64 = 0;
+
+/// Hard limit on the niceness of a process. This is inverted and the actual niceness is 20 - the
+/// rlimit. We only support one hart
+const RLIMIT_NICE: u64 = 20;
+
+/// Hard limit on the real time priority of a process. We only support one hart.
+const RLIMIT_RTPRIO: u64 = 0;
+
+/// Hard limit on CPU time without making a blocking system call. We only support one hart.
+const RLIMIT_RTTIME: u64 = u64::MAX;
+
+pub struct Rlimit<MC: MemoryConfig>(pub u64, PhantomData<MC>);
+
+impl<MC: MemoryConfig> fmt::Debug for Rlimit<MC> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        self.0.fmt(f)
+    }
+}
+
+// Resource limits for system calls
+impl<MC: MemoryConfig> TryFrom<u64> for Rlimit<MC> {
+    type Error = Error;
+
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        let value = value.try_into()?;
+
+        struct MemorySize<MC>(MC);
+        impl<MC: MemoryConfig> MemorySize<MC> {
+            const SIZE: u64 = MC::TOTAL_BYTES as u64;
+
+            const LOWER_BOUND: () = assert!(MC::TOTAL_BYTES >= u64::MIN as usize);
+            const UPPER_BOUND: () = assert!(MC::TOTAL_BYTES <= u64::MAX as usize);
```

Comment:
Out of curiousity, do these actually evaluate at compile time, or are the expressions simply inlined?